packages/component-library-react/src/FormFieldTextbox.tsx

```diff
@@ -0,0 +1,148 @@
+import { FormFieldErrorMessage } from './FormFieldErrorMessage';
+import { FormLabel } from './FormLabel';
+import { Textbox, TextboxProps } from './Textbox';
+import { TextboxTypes } from './Textbox';
+
+export interface FormFieldTextboxProps
+  extends Omit<FormFieldProps, 'type'>,
+    Pick<
+      TextboxProps,
+      | 'autoComplete'
+      | 'defaultValue'
+      | 'disabled'
+      | 'inputRequired'
+      | 'invalid'
+      | 'list'
+      | 'max'
+      | 'maxLength'
+      | 'min'
+      | 'minLength'
+      | 'name'
+      | 'pattern'
+      | 'placeholder'
+      | 'readOnly'
+      | 'required'
+      | 'size'
+      | 'step'
+      | 'value'
+      | 'type'
+    > {
+  description?: ReactNode;
+  errorMessage?: ReactNode;
+  inputDir?: TextboxProps['dir'];
+  inputRef?: Ref<HTMLInputElement>;
+  label: ReactNode;
+  status?: ReactNode;
+}
+
+export const FormFieldTextbox = forwardRef(
+  (
+    {
+      name,
+      invalid,
+      disabled,
+      label,
+      errorMessage,
+      description,
+      readOnly,
+      status,
+      autoComplete,
+      list,
+      min,
+      max,
+      minLength,
+      maxLength,
+      step,
+      placeholder,
+      pattern,
+      required,
+      inputRequired,
+      inputDir,
+      type,
+      value,
+      onChange,
+      onInput,
+      onFocus,
+      onBlur,
+      defaultValue,
+      size,
+      children,
+      inputRef,
+      ...props
+    }: PropsWithChildren<FormFieldTextboxProps>,
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => {
+    const inputId = useId();
+    const descriptionId = useId();
+    const statusId = useId();
+    const errorMessageId = useId();
+
+    return (
+      <FormField invalid={invalid} ref={ref} {...props}>
+        <div className="utrecht-form-field__label">
+          <FormLabel htmlFor={inputId}>{label}</FormLabel>
+        </div>
+        {description && (
+          <FormFieldDescription className="utrecht-form-field__description" id={descriptionId}>
+            {description}
+          </FormFieldDescription>
+        )}
+        {invalid && errorMessage && (
+          <FormFieldErrorMessage className="utrecht-form-field__error-message" id={errorMessageId}>
+            {errorMessage}
+          </FormFieldErrorMessage>
+        )}
+        <div className="utrecht-form-field__input">
+          <Textbox
+            ref={inputRef}
+            id={inputId}
+            name={name}
+            type={(type as TextboxTypes) || 'text'}
+            autoComplete={autoComplete}
+            aria-describedby={
+              clsx({
+                [descriptionId]: description,
+                [errorMessageId]: invalid,
+                [statusId]: status,
+              }) || undefined
+            }
+            invalid={invalid}
+            dir={inputDir || 'auto'}
+            disabled={disabled}
+            min={typeof min === 'number' ? String(min) : typeof min === 'string' ? min : undefined}
```

Comment:
Hmm I wouldn't expect this check to be necessary and if it is then I would expect it to be handled by the Textbox component which is using the same types?

Reply:
Probably I copied this from the stencil implementation of `<utrecht-form-field-textbox>`, but in React this isn't necessary.